### PR TITLE
Disable foreign key checks before purge on MySqlPlatform

### LINKED ISSUE #121: Truncate with foreign key check ignore
When `--purge-with-truncate` options is used, foreign key check should be deactivated.

If not, most of projects will get this kind of error:

```
SQLSTATE[42000]: Syntax error or access violation: 1701 Cannot truncate a table referenced in a foreign key constraint
```

### PR DESCRIPTION
**Pull request by vincentchalamon**
Fix #121

### CODE REVIEW

File: src/Doctrine/DataFixtures/Executor/ORMExecutor.php [hunk 1]
```diff
@@ -47,6 +48,23 @@ public function __construct(EntityManagerInterface $manager, LoaderInterface $lo
...
      */
+    public function purge()
+    {
+        $connection = $this->getObjectManager()->getConnection();
+        $mysqlPlatform = $this->purger->getPurgeMode() === ORMPurger::PURGE_MODE_TRUNCATE
+            && $connection->getDatabasePlatform() instanceof MySqlPlatform;
+        if ($mysqlPlatform) {
+            $connection->exec('SET FOREIGN_KEY_CHECKS = 0;');
+        }
+        parent::purge();
```
Comment (theofidry) [inline src/Doctrine/DataFixtures/Executor/ORMExecutor.php]: I would just add a line return before and after the `parent` to make it a bit more obvious  (I missed it at first glance)

**Comment by theofidry** [comment]
Nice one! Do you want to add checks against MySQL versions or it's "not doable" with Travis?

**Comment by vincentchalamon** [comment]
I'll try to update travis configuration in this project, using different databases like sqlite & multiple versions of mysql (check [this documentation](https://docs.travis-ci.com/user/database-setup/))

**Comment by vincentchalamon** [comment]
@theofidry What do you think of [this configuration](https://core.trac.wordpress.org/attachment/ticket/30462/30462.3.diff) ?

**Comment by vincentchalamon** [comment]
Apparently not working :( I can't find a solution to configure travis using multiple versions of MySQL…

**Comment by theofidry** [comment]
Oh, that's a pity it looked promising...

**Comment by vincentchalamon** [comment]
Well, if you find a solution, we could try it

**Comment by theofidry** [comment]
Ah, that's true :p

Well I would like to think Doctrine ORM does a better job than me at testing DB specific things^^ I do not have lots of time to play around with DB things, so if I do it must wait this weekend :/

**Comment by vincentchalamon** [comment]
I think it should be tested separately, isn't it ?

**Comment by theofidry** [comment]
Yeah true, we could get this merge and test it, without doing a new release right away

**Comment by vincentchalamon** [comment]
Thanks @theofidry
I'll try to find some time to think about how to test it (unit & functional) using each common platforms: MongoDB, Maria, MySQL, SQLite, Postgre…

**Comment by theofidry** [comment]
That would be awesome :dancers:

**Comment by vincentchalamon** [comment]
@theofidry I've a problem: I need a PR in DoctrineBundle I've made on last august, but nobody answers in Doctrine community…
https://github.com/doctrine/DoctrineBundle/pull/456

**Comment by theofidry** [comment]
Ahh gosh, *that* is what I really hate about Doctrine things, it always takes ages to get a feedback if you get any. Maybe we could try to re-ping the whole doctrine team? I'm not really for spam usually but...

**Comment by vincentchalamon** [comment]
I hate spamming too, but I already ask them for review a lot and nobody answer… That's why I hate Doctrine :disappointed: 
Maybe through Twitter ? Not sure about that solution neither…

**Comment by theofidry** [comment]
Worth trying...

**Comment by vincentchalamon** [comment]
*fingers_crossed*
https://twitter.com/vincentchalamon/status/702485698813816832

**Pull request merged**